packages/data/src/index.js

```diff
@@ -80,8 +80,9 @@ export { plugins };
 export { combineReducers };
 
 /**
- * Given the name or descriptor of a registered store, returns an object of the store's selectors.
- * The selector functions are been pre-bound to pass the current state automatically.
+ * Given a store descriptor (also supports legacy calling convention accepting the name of a
+ * registered store), returns an object of the store's selectors. The selector functions are
+ * been pre-bound to pass the current state automatically.
```

Comment:
given that we want to discourage use of passing the string I feel like we could de-emphasize explaining the legacy usage and move it either to the end of the description here, or as an additional note inside the JSDoc area

```ts
/**
 * Given a store descriptor returns an object of the store's selectors.
 * The selector functions are pre-bound to pass the current state automatically.
 *
 * Note: The legacy calling convention of passing the store name is also supported.
 *
 * @example
 * …
 */
```

Reply:
thanks for rearranging those types. although it doesn't change anything from a TypeScript parameter, I do think you made it clearer still, and I had almost suggested it but didn't want to bog down this change.